TF_Service_dotNet/TouchFree Service/Connection/Services/WebSocketReceiver.cs

```diff
@@ -1,21 +1,30 @@
-﻿using System.Collections;
-using System.Collections.Generic;
-using UnityEngine;
+﻿using System.Collections.Generic;
 using System.Collections.Concurrent;
-using Ultraleap.TouchFree.Service.ServiceTypes;
-using Ultraleap.TouchFree.ServiceShared;
 
 using Newtonsoft.Json;
 using Newtonsoft.Json.Linq;
 
-namespace Ultraleap.TouchFree.Service
+using Ultraleap.TouchFree.Library.Configuration;
+using Ultraleap.TouchFree.Service.ConnectionTypes;
+
+namespace Ultraleap.TouchFree.Service.Connection
 {
-    [DisallowMultipleComponent]
-    public class WebSocketReceiver : MonoBehaviour
+    public class WebSocketReceiver
     {
         public ConcurrentQueue<string> configChangeQueue = new ConcurrentQueue<string>();
         public ConcurrentQueue<string> configStateRequestQueue = new ConcurrentQueue<string>();
 
+        private readonly UpdateBehaviour updateBehaviour;
+        private readonly ClientConnectionManager clientMgr;
+
+        public WebSocketReceiver(UpdateBehaviour _updateBehaviour, ClientConnectionManager _clientMgr)
```

Comment:
Is this ever constructed?

Reply:
I noticed that but how is that given an UpdateBehaviour or a ClientConnectionManager? Is that because they are singletons?

Reply:
Yeah so if something being constructed as a Middleware or service has something in its constructor, it will look to the existing Singleton services to pass in. This is why the initialisation of the WSReciever has to be the _last_ of the services added in the Services list, cos it depends on the other ones.